### PR TITLE
Feat: Add New Plugin `devlake-config`

### PR DESCRIPTION
**Pull request by daniel-hutao**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description

Add new plugin `devlake-config` to config DevLake.

@warren830 has tested this pr and everything works fine.

## Related Issues

N/A

## New Behavior (screenshots if needed)

DevLake connection configuration support.

### TODO

- [ ] Delete/Update logic: code together with @warren830 
- [ ] Document: write in the next pr @daniel-hutao

### CODE REVIEW

File: internal/pkg/plugin/devlakeconfig/devlakeconfig.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			return err
+		}
+	}
+
+	log.Infof("All %s connections have been created.", pluginName)
+	return nil
+}
+
+func createConnection(url string, bodyWithJson []byte) error {
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(bodyWithJson))
```
Comment (iyear) [inline internal/pkg/plugin/devlakeconfig/devlakeconfig.go:58]: `http.MethodPost`

File: internal/pkg/plugin/devlakeconfig/devlakeconfig.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createConnections(host string, pluginName string, connections []Connection) error {
+	for _, c := range connections {
+		log.Infof("(%s)", c.Name)
```
Comment (iyear) [inline internal/pkg/plugin/devlakeconfig/devlakeconfig.go:37]: Maybe more text is needed

File: internal/pkg/plugin/devlakeconfig/devlakeconfig.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+func ApplyConfig(options plugininstaller.RawOptions) error {
+	opts, err := NewOptions(options)
+	if err != nil {
+		return err
+	}
+
+	for _, p := range opts.Plugins {
+		log.Infof("Got DevLake plugin config: %s. Connections: ", p.Name)
```
Comment (iyear) [inline internal/pkg/plugin/devlakeconfig/devlakeconfig.go:46]: missing `Connections: balabala`

**Review by iyear — APPROVED** [review]
LGTM

**Review by warren830 — APPROVED** [review]
LGTM

File: internal/pkg/plugin/devlakeconfig/devlakeconfig.go (hunk 3 above)
Comment (daniel-hutao) [reply internal/pkg/plugin/devlakeconfig/devlakeconfig.go:46]: image

The connection logs is followed below. @iyear cc @warren830

**Pull request merged**